### PR TITLE
BOT API 7.2

### PR DESCRIPTION
**Pull request by coder2020official**
Not promising anything, let's see how far I can go.

Bot API 7.2

Integration with Business Accounts

- [x] Added the class [BusinessConnection](https://core.telegram.org/bots/api#businessconnection) and updates about the connection or disconnection of the bot to a business account, represented by the field business_connection in the class [Update](https://core.telegram.org/bots/api#update).

- [x] Added updates about new messages in a business account connected to the bot, represented by the field business_message in the class [Update](https://core.telegram.org/bots/api#update).
- [x] Added updates about message edits in a business account connected to the bot, represented by the field edited_business_message in the class [Update](https://core.telegram.org/bots/api#update).
- [x] Added updates about message deletion in a business account connected to the bot, represented by the class [BusinessMessagesDeleted](https://core.telegram.org/bots/api#businessmessagesdeleted) and the field deleted_business_messages in the class [Update](https://core.telegram.org/bots/api#update).
- [x] Added the method [getBusinessConnection](https://core.telegram.org/bots/api#getbusinessconnection).
Working on Behalf of Business Accounts

- [x] Added the parameter business_connection_id to the methods [sendMessage](https://core.telegram.org/bots/api#sendmessage), [sendPhoto](https://core.telegram.org/bots/api#sendphoto), [sendVideo](https://core.telegram.org/bots/api#sendvideo), [sendAnimation](https://core.telegram.org/bots/api#sendanimation), [sendAudio](https://core.telegram.org/bots/api#sendaudio), [sendDocument](https://core.telegram.org/bots/api#senddocument), [sendSticker](https://core.telegram.org/bots/api#sendsticker), [sendVideoNote](https://core.telegram.org/bots/api#sendvideonote), [sendVoice](https://core.telegram.org/bots/api#sendvoice), [sendLocation](https://core.telegram.org/bots/api#sendlocation), [sendVenue](https://core.telegram.org/bots/api#sendvenue), [sendContact](https://core.telegram.org/bots/api#sendcontact), [sendPoll](https://core.telegram.org/bots/api#sendpoll), [sendDice](https://core.telegram.org/bots/api#senddice), [sendGame](https://core.telegram.org/bots/api#sendgame), and [sendMediaGroup](https://core.telegram.org/bots/api#sendmediagroup).
- [x] Added the parameter business_connection_id to the method [sendChatAction](https://core.telegram.org/bots/api#sendchataction).
- [x] Added the field business_connection_id to the class [Message](https://core.telegram.org/bots/api#message).
- [x] Added the field sender_business_bot to the class [Message](https://core.telegram.org/bots/api#message).
Information about Business Accounts

- [x] Added the class [BusinessIntro](https://core.telegram.org/bots/api#businessintro) and the field business_intro to the class [Chat](https://core.telegram.org/bots/api#chat).
- [x] Added the class [BusinessLocation](https://core.telegram.org/bots/api#businesslocation) and the field business_location to the class [Chat](https://core.telegram.org/bots/api#chat).
- [x] Added the classes [BusinessOpeningHours](https://core.telegram.org/bots/api#businessopeninghours) and [BusinessOpeningHoursInterval](https://core.telegram.org/bots/api#businessopeninghoursinterval) and the field business_opening_hours to the class [Chat](https://core.telegram.org/bots/api#chat).
Mixed-Format Sticker Packs

- [ ] Removed the fields is_animated and is_video from the class [StickerSet](https://core.telegram.org/bots/api#stickerset).
- [x] Added the field format to the class [InputSticker](https://core.telegram.org/bots/api#inputsticker).
- [ ] Removed the parameter sticker_format from the method [createNewStickerSet](https://core.telegram.org/bots/api#createnewstickerset).
- [x] Added the parameter format to the method [setStickerSetThumbnail](https://core.telegram.org/bots/api#setstickersetthumbnail).
- [x] Increased the maximum number of stickers in any regular and mask sticker set to 120.
- [x] Allowed to upload WEBM stickers using [SendSticker](https://core.telegram.org/bots/api#sendsticker).
Request Chat Improvements

- [x] Added the fields request_name, request_username, and request_photo to the class [KeyboardButtonRequestUsers](https://core.telegram.org/bots/api#keyboardbuttonrequestusers).
- [x] Added the fields request_title, request_username, and request_photo to the class [KeyboardButtonRequestChat](https://core.telegram.org/bots/api#keyboardbuttonrequestchat).
- [x] Added the class SharedUser and replaced the field user_ids in the class [UsersShared](https://core.telegram.org/bots/api#usersshared) with the field users.
- [x] Added the fields title, username, and photo to the class [ChatShared](https://core.telegram.org/bots/api#chatshared).
Other Changes

- [x] Added the field is_from_offline to the class [Message](https://core.telegram.org/bots/api#message).
- [x] Added the field can_connect_to_business to the class [User](https://core.telegram.org/bots/api#user).
- [x] Added the field personal_chat to the class [Chat](https://core.telegram.org/bots/api#chat).
- [x] Added the method [replaceStickerInSet](https://core.telegram.org/bots/api#replacestickerinset),
- [x] Added the class [Birthdate](https://core.telegram.org/bots/api#birthdate) and the field birthdate to the class [Chat](https://core.telegram.org/bots/api#chat).
- [x] Added the field BiometricManager to the class [WebApp](https://core.telegram.org/bots/webapps#initializing-mini-apps).

### CODE REVIEW

**Comment by coder2020official** [comment]
- Skipping:
- Removed the fields is_animated and is_video from the class [StickerSet](https://core.telegram.org/bots/api#stickerset).
- Removed the parameter sticker_format from the method [createNewStickerSet](https://core.telegram.org/bots/api#createnewstickerset).
@badiboy they made stickerpacks mixed, meaning you can have both vids, photo stickers etc; now my brain is not braining regarding the actions to keep backward compatibility

**Comment by coder2020official** [comment]
@badiboy done, need your opinion on 2 things; and your review

**Comment by Badiboy** [comment]
Got it.

**Comment by Badiboy** [comment]
@coder2020official 50% reviewed, mostly fine. Hope to finish tomorrow.

**Review by Badiboy — CHANGES_REQUESTED** [review]

**Comment by Badiboy** [comment]
@coder2020official I propose to fix this issues and merge the PR. Then continue with 3 open questions mentioned in your main topic. To avoid long live of this huge update.

File: telebot/async_telebot.py [hunk 1]
```diff
@@ -944,6 +999,23 @@ def setup_middleware(self, middleware: BaseMiddleware):
...
+    def check_commands_input(commands, method_name):
+        """
+        :meta private:
+        """
+        if not isinstance(commands, list) or not all(isinstance(item, str) for item in commands):
+            logger.error(f"{method_name}: Commands filter should be list of strings (commands), unknown type supplied to the 'commands' filter list. Not able to use the supplied type.")
+
+
+    @staticmethod
+    def check_regexp_input(regexp, method_name):
```
[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/async_telebot.py:1012]: What do you mean? I think I copied it from sync

**Comment by coder2020official** [comment]
And what are we gonna do about the things I skipped

**Comment by Badiboy** [comment]
image

**Comment by coder2020official** [comment]
And can you answer my comment

**Comment by Badiboy** [comment]
I did. Firstly we'll close this huge update. Then address other or non/closed items.

**Comment by coder2020official** [comment]
Can't see your comment on your changes:
> Used, but not defined in SyncTelebot.
> What do you mean? I think I copied it from sync

File: telebot/__init__.py [hunk 2]
```diff
@@ -5415,10 +5559,13 @@ def set_sticker_set_thumbnail(self, name: str, user_id: int, thumbnail: Union[An
         :param thumbnail: A .WEBP or .PNG image with the thumbnail, must be up to 128 kilobytes in size and have a width and height of exactly 100px, or a .TGS animation with a thumbnail up to 32 kilobytes in size (see https://core.telegram.org/stickers#animated-sticker-requirements for animated sticker technical requirements), or a WEBM video with the thumbnail up to 32 kilobytes in size; see https://core.telegram.org/stickers#video-sticker-requirements for video sticker technical requirements. Pass a file_id as a String to send a file that already exists on the Telegram servers, pass an HTTP URL as a String for Telegram to get a file from the Internet, or upload a new one using multipart/form-data. More information on Sending Files ». Animated and video sticker set thumbnails can't be uploaded via HTTP URL. If omitted, then the thumbnail is dropped and the first sticker is used as the thumbnail.
         :type thumbnail: :obj:`filelike object`
 
+        :param format: Format of the thumbnail, must be one of “static” for a .WEBP or .PNG image, “animated” for a .TGS animation, or “video” for a WEBM video
```
[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:5562]: Btw, what do I set format to? 
> correspond to previous behaviour
I think previously only .png and .webp thumbnails were supported, can't remember though; what do I do

**Comment by coder2020official** [comment]
Oops, I forgot to submit my comments. Check now

**Comment by Badiboy** [comment]
> Can't see your comment on your changes:
> 
> > Used, but not defined in SyncTelebot.
> > What do you mean? I think I copied it from sync

You are right. I checked only in "modified" and it was not visible. Sorry.

**Comment by coder2020official** [comment]
So I ignore your 2 comments?

File: telebot/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:5562]: "None" as default parameter value.
"static" + deprecation warning - if the value is not passed

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:5562]: got it

[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:5562]: If format is passed - use format.
elif is_animated=True - “animated” and deprecation warning
elif is_video=True - “video” and deprecation warning
else - “static” and deprecation warning

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:5562]: but there are no such params?

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:5562]: I think you were correct the first time

[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:5562]: > but there are no such params?
Hmmm... Where I saw it... Oky then. Let's be static if undefined.

**Comment by coder2020official** [comment]
I think I am done

**Comment by Badiboy** [comment]
🤝 Let's merge and then go to remaining items.

**Pull request merged**